### ISSUE TITLE
[Bug] Path rendering crashing on GeometryExtensions due to unexpected PathSegment configuration

### ISSUE THREAD

**Issue by MitchBomcanhao** [xamarin/Xamarin.Forms#11653]
### Description
It appears to be possible to make a Path that has a segment with a number of points that will cause the app to crash.

example on Android: If the PolyBezierSegment has a small enough number of points, it will fail.
```
// PolyBezierSegment
else if (pathSegment is PolyBezierSegment)
{
    PolyBezierSegment polyBezierSegment = pathSegment as PolyBezierSegment;
    PointCollection points = polyBezierSegment.Points;

    for (int i = 0; i < points.Count; i += 3)
    {
        path.CubicTo(
            density * (float)points[i + 0].X, density * (float)points[i + 0].Y,
            density * (float)points[i + 1].X, density * (float)points[i + 1].Y,
            density * (float)points[i + 2].X, density * (float)points[i + 2].Y);
    }

    lastPoint = points[points.Count - 1];
}
```
There are other types of segments that will have similar failures.
This issue may depend per platform as each one has a slightly different implementation for each segment type.

### Steps to Reproduce

1.  Create a Path with something like this
```
<Path Aspect="Uniform">
    <Path.Data>
        <PathGeometry>
            <PathGeometry.Figures>
                <PathFigureCollection>
                    <PathFigure IsClosed="True" StartPoint="10,100">
                        <PathFigure.Segments>
                            <PathSegmentCollection>
                                <PolyBezierSegment Points="100,100" />
                            </PathSegmentCollection>
                        </PathFigure.Segments>
                    </PathFigure>
                </PathFigureCollection>
            </PathGeometry.Figures>
        </PathGeometry>
    </Path.Data>
</Path>
```

### Expected Behavior
Not sure, should it be blank because the path is not as expected?

### Actual Behavior
App crashes with out of range exception

### Basic Information
- Version with issue: 4.7.0.1239
- Last known good version: n/a

**Comment by PureWeen** [comment]
@jsuarezruiz  FYI

**Comment by samhouts** [comment]
closed by #11873
